workspaces/diff-engine/cli/src/commit.rs

```diff
@@ -0,0 +1,185 @@
+use super::events_from_chunks;
+use chrono::Utc;
+use clap::{App, Arg, ArgMatches, SubCommand};
+use futures::StreamExt;
+use optic_diff_engine::streams;
+use optic_diff_engine::Aggregate;
+use optic_diff_engine::CommandContext;
+use optic_diff_engine::{RfcCommand, SpecCommand, SpecCommandHandler};
+use optic_diff_engine::{SpecChunkEvent, SpecEvent};
+use std::path::Path;
+use std::process;
+use tokio::io::{stdin, stdout};
+use uuid::Uuid;
+
+pub const SUBCOMMAND_NAME: &'static str = "commit";
+
+pub fn create_subcommand<'a, 'b>() -> App<'a, 'b> {
+  SubCommand::with_name(SUBCOMMAND_NAME)
+    .about("Commits results of commands received over stdin as a new batch commit")
+    .arg(
+      Arg::with_name("commit-message")
+        .short("m")
+        .required(true)
+        .value_name("COMMIT_MESSAGE")
+        .takes_value(true)
+        .help("The commit message describing the commands as a whole"),
+    )
+    .arg(
+      Arg::with_name("append-to-root")
+        .long("append-to-root")
+        .required(false)
+        .takes_value(false)
+        .help("Append new batch commit to the root spec file, instead of a new spec change file"),
+    )
+    .arg(
+      Arg::with_name("client-session-id")
+        .long("client-session-id")
+        .required(false)
+        .value_name("CLIENT_SESSION_ID")
+        .takes_value(true)
+        .default_value("unknown-session")
+        .help("The session id of the client requesting the commands to be committed"),
+    )
+    .arg(
+      Arg::with_name("client-id")
+        .long("client-id")
+        .required(false)
+        .value_name("CLIENT_ID")
+        .takes_value(true)
+        .default_value("anonymous")
+        .help("Unique id of the client the commands to be committed"),
```

Comment:
With manual testing, instantly ran into requiring these contexts being a pain, so defined default values for them. We _can_ still require them at other layers if we want users to always provide them, but the CLI should only be developer used at the moment.